libs/ui/switch/helm/src/lib/hlm-switch.component.ts

```diff
@@ -0,0 +1,106 @@
+import { hlm } from '@spartan-ng/ui-core';
+import { BrnSwitchComponent, BrnSwitchThumbComponent } from '@spartan-ng/ui-switch-brain';
+import { ClassValue } from 'clsx';
+import { HlmSwitchThumbDirective } from './hlm-switch-thumb.directive';
+
+export const HLM_SWITCH_VALUE_ACCESSOR = {
+	provide: NG_VALUE_ACCESSOR,
+	useExisting: forwardRef(() => HlmSwitchComponent),
+	multi: true,
+};
+
+@Component({
+	selector: 'hlm-switch',
+	imports: [BrnSwitchThumbComponent, BrnSwitchComponent, HlmSwitchThumbDirective],
+	standalone: true,
+	host: {
+		'[attr.class]': 'null',
+		'[attr.id]': 'null',
+		'[attr.aria-label]': 'null',
+		'[attr.aria-labelledby]': 'null',
+		'[attr.aria-describedby]': 'null',
+	},
+	template: `
+		<brn-switch
+			[class]="_computedClass()"
+			[checked]="_checked()"
+			(changed)="handleChange($event)"
+			(touched)="_onTouched()"
+			[id]="id"
+			[aria-label]="ariaLabel"
+			[aria-labelledby]="ariaLabelledby"
+			[aria-describedby]="ariaDescribedby"
+		>
+			<brn-switch-thumb hlm />
+		</brn-switch>
+	`,
+	providers: [HLM_SWITCH_VALUE_ACCESSOR],
+})
+export class HlmSwitchComponent {
+	private readonly _userCls = signal<ClassValue>('');
+	@Input()
+	set class(userCls: ClassValue) {
+		this._userCls.set(userCls);
+	}
+
+	@Output()
+	public changed = new EventEmitter<boolean>();
+
+	handleChange(value: boolean): void {
+		const previousChecked = this._checked();
+		this._checked.set(value);
+		this._onChange(!previousChecked);
+		this.changed.emit(!previousChecked);
+	}
+
+	/** CONROL VALUE ACCESSOR */
+	protected _checked = signal(false);
+	@Input({ transform: booleanAttribute })
+	set checked(value: boolean) {
+		this._checked.set(value);
+	}
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	writeValue(value: any): void {
+		this.checked = !!value;
+	}
+	// eslint-disable-next-line @typescript-eslint/no-empty-function,@typescript-eslint/no-unused-vars,,@typescript-eslint/no-explicit-any
+	protected _onChange = (_: any) => {};
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
+	protected _onTouched = () => {};
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	registerOnChange(fn: any): void {
+		this._onChange = fn;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	registerOnTouched(fn: any): void {
+		this._onTouched = fn;
+	}
+
+	/** Used to set the id on the underlying brn element. */
+	@Input()
+	id: string | null = null;
+
+	/** Used to set the aria-label attribute on the underlying brn element. */
+	@Input('aria-label')
+	ariaLabel: string | null = null;
+
+	/** Used to set the aria-labelledby attribute on the underlying brn element. */
+	@Input('aria-labelledby')
+	ariaLabelledby: string | null = null;
+
+	/** Used to set the aria-describedby attribute on the underlying brn element. */
+	@Input('aria-describedby')
+	ariaDescribedby: string | null = null;
+
+	protected _computedClass = computed(() => this._generateClass());
```

Comment:
I think we can get rid of the `this._generateClass()` function and just put the hlm directly into the computed.

Reply:
My usual plan was to get rid of _generateClass for all components, once we are on 17.1 and we do the signal inputs refactoring. ;)
but i can do it for this now, as you already started to remove the extra function in other components I don't see a problem.